azurelinuxagent/daemon/scvmm.py

```diff
@@ -70,4 +71,5 @@ def run(self):
         if self.detect_scvmm_env():
             self.start_scvmm_agent()
```

Comment:
Setting `Restart=on-failure` is not working. Most probably scvmmagent is not cleanly exiting and waagent loops.